gitlabform/gitlabform/core.py

```diff
@@ -411,9 +410,20 @@ def process_secret_variables(self, project_and_group, configuration):
             return
 
         logging.debug("Secret variables BEFORE: %s", self.gl.get_secret_variables(project_and_group))
+
         for secret_variable in sorted(configuration['secret_variables']):
             logging.info("Setting secret variable: %s", secret_variable)
 
+            if "delete" in configuration['secret_variables'][secret_variable]:
+                key = configuration['secret_variables'][secret_variable]["key"]
+                if configuration['secret_variables'][secret_variable]["delete"]:
+                    logging.debug(f"Deleting {secret_variable}: {key} in project {project_and_group}")
+                    try:
+                        self.gl.delete_secret_variable(project_and_group, key)
+                    except:
+                        logging.warn(f"Could not delete variable {key} in group {project_and_group}")
+                    return
```

Comment:
I think you meant `continue`, not `return`, as we don't want to completely exit the function as we may have more secret variables to process in this `for` loop.

Reply:
This applies to the code in group variables function too.